### PR TITLE
Show stacked charts and stacking options with just one series

### LINKED ISSUE #9362: Stacked Percentage Bar Chart Breaks When Only One Series Shown
If a stacked bar chart was created when multiple series existed but due to a filter changing, a filter's value being dynamic, or data changing only one series shows the bar chart fails to function as expected. In a real world example this happens when using "past 7 days" and the past 7 days happens to no longer have data for more than one series.

## Steps To Reproduce
1. Create a question with the following data. Setup a status text filter and give it a value of "foo". Make the bar chart a stacked bar chart.
```
select * from (
select to_date('2019-01-28', 'yyyy-mm-dd') "Date", 'SUCCESS' "Status", 12 "Count" from dual
union select to_date('2019-01-28', 'yyyy-mm-dd') "Date", 'SUCCESS' "Status", 3 "Count" from dual
union select to_date('2019-01-29', 'yyyy-mm-dd') "Date", 'SUCCESS' "Status", 3 "Count" from dual
union select to_date('2019-01-30', 'yyyy-mm-dd') "Date", 'SUCCESS' "Status", 4 "Count" from dual
union select to_date('2019-01-31', 'yyyy-mm-dd') "Date", 'SUCCESS' "Status", 4 "Count" from dual
union select to_date('2019-01-31', 'yyyy-mm-dd') "Date", 'FAILURE' "Status", 1 "Count" from dual
) where "Status" != {{status}}
```
image

2. Change the filter to be "FAILURE". The bar chart will no longer be a stacked percentage bar chart

Result:
image

Your browser and the version: 71.0.3578.98 (Official Build) (64-bit)
Your operating system: Windows 7
Your databases: Oracle 12.2
Metabase version: 0.31.2
Metabase hosting environment: RHEL 7
Metabase internal database: MySQL

### PR DESCRIPTION
**Pull request by paulrosenzweig**
Resolves #9362 

**I'm not convinced this is a good change. Is the fix worse than the initial bug?**

The issue was due to a chart getting set up on data with multiple series, but later the data updates and only one series is present. I updated the behavior to keep stacking settings (including normalization) even if only one series is present.

With that change, there was a new weird scenario: the stacking settings still depend on having multiple series. While your chart will now maintain its stacking settings, there'd be no way to remove them. To prevent this, I updated the `getHidden` behavior on that setting to display if any series are stackable.

The result is that you can now turn on stacking even if you just have one series to begin with. Is that ok? I feel it might be a bit confusing.

One compromise I tried out but decided against is to have the setting's display logic rely on either having multiple series _or_ currently having stacking turned on. There, you'd be able to keep stacking even if data reduced to a single series and you'd be able to remove stacking whenever it's on. The downside of this approach is that when you removed stacking the settings would disappear.

### CODE REVIEW

**Review by kdoh — APPROVED** [review]
I think this is ok. While it might not be the first thing you'd think to do in just your average single series bar chart, it's actually a little crazy that you'd need to have multi series data present to create a chart that's capable of showing that data were it to change (as in the original issue), so I think overall this is a net win. I'd also rather always have the setting and have it be a little less useful / sensical in the single series state than have that availability be conditional on logic that's somewhat hard to parse on the part of the end user, which coincidentally also makes documentation and testing harder.

**Review by tlrobinson — APPROVED** [review]
I agree with @kdoh
